src/util/__tests__/preprocessData-test.js

```diff
@@ -0,0 +1,313 @@
+    // prettier-ignore
+    const input = [
+      {"pid":57632,"tid":38659,"ts":874860756135,"ph":"X","cat":"disabled-by-default-devtools.timeline","name":"RunTask","dur":18,"tdur":19,"tts":8700284918,"args":{}},
+      {"pid":57632,"tid":38659,"ts":874860756158,"ph":"X","cat":"disabled-by-default-devtools.timeline","name":"RunTask","dur":30,"tdur":30,"tts":8700284941,"args":{}},
+      {"pid":57632,"tid":38659,"ts":874860756192,"ph":"X","cat":"disabled-by-default-devtools.timeline","name":"RunTask","dur":21,"tdur":20,"tts":8700284976,"args":{}},
+      {"pid":57632,"tid":38659,"ts":874860756216,"ph":"X","cat":"disabled-by-default-devtools.timeline","name":"RunTask","dur":6,"tdur":5,"tts":8700285000,"args":{}},
+      {"pid":57632,"tid":38659,"ts":874860756224,"ph":"X","cat":"disabled-by-default-devtools.timeline","name":"RunTask","dur":7,"tdur":6,"tts":8700285008,"args":{}},
+      {"pid":57632,"tid":38659,"ts":874860756233,"ph":"X","cat":"disabled-by-default-devtools.timeline","name":"RunTask","dur":5,"tdur":4,"tts":8700285017,"args":{}},
+    ]
+
+    expect(preprocessData(input)).toEqual({
+      startTime: 874860756135,
+      duration: 0,
+      events: [],
+      measures: [],
+    });
+  });
+
+  it('should throw if unrecognized React mark is encountered', () => {
+    expect(() =>
+      preprocessData([
+        // prettier-ignore
+        {"args":{"data":{"navigationId":"E082C30FBDA3ACEE0E7B5FD75F8B7F0D"}},"cat":"blink.user_timing","name":"--there-are-four-lights","ph":"R","pid":17232,"tid":13628,"ts":264686513020,"tts":4082554},
+      ])
+    ).toThrow();
+  });
+
+  it('should throw if events and measures are incomplete', () => {
+    const error = jest.spyOn(console, 'error');
+    preprocessData([
+      // prettier-ignore
+      {"args":{"data":{"navigationId":"1065756F5FDAD64BE45CA86B0BBC1F8B"}},"cat":"blink.user_timing","name":"--render-start-8","ph":"R","pid":1852,"tid":12484,"ts":42351664678,"tts":1512475},
+    ]);
+    expect(error).toHaveBeenCalled();
+  });
+
+  it('should throw if work is completed without being started', () => {
+    const error = jest.spyOn(console, 'error');
+    preprocessData([
+      // prettier-ignore
+      {"args":{"data":{"navigationId":"E082C30FBDA3ACEE0E7B5FD75F8B7F0D"}},"cat":"blink.user_timing","name":"--render-yield","ph":"R","pid":17232,"tid":13628,"ts":264686513020,"tts":4082554},
+    ]);
+    expect(error).toHaveBeenCalled();
+
+    // TODO: add others
+  });
+
+  it('should process complete set of events (page load sample data)', () => {
+    expect(
+      // prettier-ignore
+      preprocessData([
+        {"args":{"data":{"documentLoaderURL":"https://concurrent-demo.now.sh/","isLoadingMainFrame":true,"navigationId":"43BC238A4FB7548146D3CD739C9C9434"},"frame":"FD65D9AFD04B1295CEA36B883F0FA82F"},"cat":"blink.user_timing","name":"navigationStart","ph":"R","pid":9312,"tid":10252,"ts":8993749139,"tts":1646191},
```

Comment:
This array is extremely long and it may be a good idea to put it in a separate JSON file, but I think there's some value in having the data visible when you read the tests, instead of having to poke in another file. Opinions welcome but I think this inlined array is good enough for now.